assimilation_code/modules/io/io_filenames_mod.f90

```diff
@@ -568,9 +567,12 @@ subroutine check_correct_variables(netcdf_filename, dom)
    ! check number of dimensions are the same.  we are comparing
    ! the full number of dimensions including time and member if present.
    if (ndims /= get_io_num_dims(dom,i)) then
-      write(msgstring,*) 'ndims ', get_io_num_dims(dom,i), ' in state does not', &
-                         ' match ndims ', ndims, ' in ', trim(netcdf_filename)
-      call error_handler(E_ERR, 'check_correct_variables', msgstring, source)
+      write(string1,*)'domain ',dom, ' variable ',trim(get_variable_name(dom,i))
+      write(string2,*)'ndims ', get_io_num_dims(dom,i), ' in state does not', &
+                      ' match ndims ', ndims
+      write(string3,*) ' in "'//trim(netcdf_filename)//'"'
+      call error_handler(E_ERR, 'check_correct_variables', string1, &
+                         source, text2=string2, text3=string3)
```

Comment:
With multiple domains, it is very nice to know which domain (and variable) is inconsistent.